### ISSUE TITLE
GPU error codes

### ISSUE THREAD

**Issue by janden** [flatironinstitute/finufft#284]
So it looks like internally, we don't really use the error codes much:
```
$ git grep return -- src/cuda | head -n 20
src/cuda/1d/cufinufft1d.cu:            return ier;
src/cuda/1d/cufinufft1d.cu:    return ier;
src/cuda/1d/cufinufft1d.cu:            return ier;
src/cuda/1d/cufinufft1d.cu:    return ier;
src/cuda/1d/interp1d_wrapper.cu:            return ier;
src/cuda/1d/interp1d_wrapper.cu:            return ier;
src/cuda/1d/interp1d_wrapper.cu:    return ier;
src/cuda/1d/interp1d_wrapper.cu:                return 1;
src/cuda/1d/interp1d_wrapper.cu:        return 2;
src/cuda/1d/interp1d_wrapper.cu:    return ier;
src/cuda/1d/interp1d_wrapper.cu:    return 0;
src/cuda/1d/spread1d_wrapper.cu:            return ier;
src/cuda/1d/spread1d_wrapper.cu:            return ier;
src/cuda/1d/spread1d_wrapper.cu:    return ier;
src/cuda/1d/spread1d_wrapper.cu:            return 1;
src/cuda/1d/spread1d_wrapper.cu:            return 1;
src/cuda/1d/spread1d_wrapper.cu:        return 2;
src/cuda/1d/spread1d_wrapper.cu:    return ier;
src/cuda/1d/spread1d_wrapper.cu:            return 1;
src/cuda/1d/spread1d_wrapper.cu:    return 0;
```
I'm guessing some of these inherit error codes from the Finufft utils, but still, it's not a great sign. Will have to go through these and see whether they're returning the error codes they should be (according to our definitions, returning 1 should be a warning that epsilon is too small, which I don't think is what's happening in the above examples).

**Comment by janden** [comment]
I'd vote for still returning an error (maybe 100 as a placeholder). If we change everything to returning zero, we'd have to figure out later on which zeros are supposed to be zero and which are actual errors…

**Comment by janden** [comment]
Closed by #348.
